### ISSUE TITLE
Make Entry behavior more consistent

### ISSUE THREAD

**Issue by bassosimone** [measurement-kit/measurement-kit#740]
Entry was originally meant to be a full fledged class masking all of `nlohmann::json` until we realized that we would live only once and it would perhaps be better to just inherit directly from `nlohmann::json`. However, this means that some methods of Entry do wrap the methods of `nlohmann::json` and convert exceptions, while others don't. In the interest of consistency, I'd say it would be better to avoid wrapping and just use `nlohmann::json` directly.

Ideally, if possible, it should be:

``` C++
using Entry = nlohmann::json;
```

**Comment by bassosimone** [comment]
Postponed to v0.4.0

**Comment by bassosimone** [comment]
This has been finally implemented!
